agency/general-agent.go

```diff
@@ -141,6 +144,7 @@ func (agentState *GeneralAgentInfo) getSystemMessage() (*engines.Message, error)
 		Content: contextString,
 		ID:      &messageId,
 	}
+	agentState.historyAppenderChannel <- systemMessage
```

Comment:
it should send it to systemWriterChannel, right?

Reply:
Yes, that is indeed what it was supposed to be...saving error on my part most likely